### PR TITLE
fix(locators): reject the race if there are only failures

### PR DESCRIPTION
**Pull request by OrKoN**

### CODE REVIEW

File: test/src/locator.spec.ts [hunk 1]
```diff
@@ -222,7 +222,9 @@ describe('Locator', function () {
     });
 
     it('should time out', async () => {
-      const clock = sinon.useFakeTimers();
+      const clock = sinon.useFakeTimers({
+        shouldClearNativeTimers: true,
```
Comment (OrKoN) [inline test/src/locator.spec.ts:226]: the lack of this config was causing hanging after all tests: apparently, the default behaviour for sinon is not to invoke the native clearTimeout if the timer was started before the fake timers were activated.

File: packages/puppeteer-core/src/api/Locator.ts [hunk 2]
```diff
@@ -732,6 +732,26 @@ class RaceLocatorImpl extends Locator {
...
+    if (
+      results.some(result => {
+        return result.status === 'fulfilled';
+      })
+    ) {
+      return;
+    }
+
+    for (const result of results) {
+      if (result.status === 'fulfilled') {
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/api/Locator.ts]: How can we get in this block at all?

Comment (OrKoN) [reply packages/puppeteer-core/src/api/Locator.ts]: currently no, but TypeScript does not know

**Review by Lightning00Blade — APPROVED** [review]

File: packages/puppeteer-core/src/api/Locator.ts (hunk 2 above)
Comment (OrKoN) [reply packages/puppeteer-core/src/api/Locator.ts]: updated

**Pull request merged**